### PR TITLE
Change node labels in h2o-3 nightly Pipeline

### PR DESCRIPTION
**Pull request by DinukaH2O**
### Description

Change node label in h2o-3-nightly-pipeline and Stick on to DC machines (a subset of machines) only so that others can use the K8s cluster used for dynamic Jenkins agents.

Changed labels to : DC && linux && docker

### CODE REVIEW

**Comment by valenad1** [comment]
You are testing it on Nightly and after that we will change all Jenkinsfiles? All beside release, we run it in master on purpose.

File: scripts/jenkins/jenkinsfiles/Jenkinsfile-Nightly [hunk 1]
```diff
@@ -1,6 +1,6 @@
 @Library('test-shared-library') _
 
-def DEFAULT_NODE_LABEL = 'h2o-3 && docker && !mr-0xc8 && (!micro || micro_21)'
+def DEFAULT_NODE_LABEL = 'DC && linux && docker'
```
Comment (mn-mikke) [inline scripts/jenkins/jenkinsfiles/Jenkinsfile-Nightly:3]: Shouldn't the `h2o-3` label be there? I remember Devops team complaining that we're using to many nodes and other projects starve

Comment (mn-mikke) [reply scripts/jenkins/jenkinsfiles/Jenkinsfile-Nightly:3]: Btw, what does `DC` mean?

**Comment by DinukaH2O** [comment]
We allocated some machines with the h2o-3 labels
http://jenkins.h2o.local:8080/label/h2o-3/

And changed the node label in all Jenkins files to h2o-3 

Let me know how it goes!

File: scripts/jenkins/jenkinsfiles/Jenkinsfile-Nightly (hunk 1 above)
Comment (valenad1) [reply scripts/jenkins/jenkinsfiles/Jenkinsfile-Nightly]: @DinukaH2O

**Comment by valenad1** [comment]
Merged to master, if it goes well, we merge it to release branch also

**Pull request merged**